stl/inc/xutility

```diff
@@ -3444,20 +3514,29 @@ public:
     static constexpr bool _Unwrap_when_unverified = _Do_unwrap_when_unverified_v<iterator_type>;
 
     template <class _Src, enable_if_t<_Wrapped_seekable_v<iterator_type, const _Src&>, int> = 0>
-    constexpr void _Seek_to(const move_iterator<_Src>& _It) {
+    constexpr void _Seek_to(const move_iterator<_Src>& _It) noexcept(noexcept(_Current._Seek_to(_It._Get_current()))) {
         _Current._Seek_to(_It.base());
     }
     template <class _Src, enable_if_t<_Wrapped_seekable_v<iterator_type, _Src>, int> = 0>
-    constexpr void _Seek_to(move_iterator<_Src>&& _It) {
-        _Current._Seek_to(_STD move(_It).base());
+    constexpr void _Seek_to(move_iterator<_Src>&& _It) noexcept(
+        noexcept(_Current._Seek_to(_STD move(_It)._Get_current()))) {
+        _Current._Seek_to(_STD move(_It)._Get_current());
+    }
+
+    constexpr const iterator_type& _Get_current() const& noexcept {
+        return _Current;
+    }
+    constexpr iterator_type&& _Get_current() && noexcept {
+        return static_cast<iterator_type&&>(_Current);
     }
 
 private:
     iterator_type _Current{};
 };
 
 template <class _Iter1, class _Iter2>
-_NODISCARD _CONSTEXPR17 bool operator==(const move_iterator<_Iter1>& _Left, const move_iterator<_Iter2>& _Right)
+_NODISCARD _CONSTEXPR17 bool operator==(const move_iterator<_Iter1>& _Left,

```

Comment:
Should these comparison operators use `_Get_current()` instead of `.base()`?

Reply:
In theory, that's observable - the current `operator==` does a copy construction and comparison on prvalues in <20. I don't know if we're allowed to do that (are we?)

Reply:
The spec notably has no "Effects" element. There's a "Returns" element that says what value we have to return, but nothing that really constrains how we obtain that value. So while `x.base() == y.base()` is conforming, I believe `x._Get_current() == y._Get_current()` is conforming as well. (Both old and new iterator requirements say this expression must be valid regardless of value category and const-qualification in different ways.)